includes/utils.php

```diff
@@ -195,10 +195,19 @@ function set_meta( $post_id, $meta ) {
 				$prev_value = maybe_unserialize( $existing_meta[ $meta_key ][ $meta_placement ] );
 			}
 
+			$is_json = false;
+
 			if ( ! is_array( $meta_value ) ) {
+				json_decode( $meta_value );
+				$is_json    = json_last_error() === 0 ? true : false;
```

Comment:
I get we are trying to determine here if the value is json, just wondering if there's any edge cases that this might have issues with? For instance, seems like the value `true` will pass as valid json (which may be fine).

Also want to ensure we won't ever have PHP errors by trying to run `json_decode` on data that can't be decoded. I know passing an `array` to that will throw an error, though that isn't a risk here because of the `if` statement. But are there any other types of data that might cause problems here?

Reply:
Also might be worth having a code comment here to explain what we're doing. I could see this confusing future developers trying to understand what's going on